concordium-consensus/src/Concordium/KonsensusV1/Consensus.hs

```diff
@@ -0,0 +1,228 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Concordium.KonsensusV1.Consensus where
+
+import Control.Monad.State
+import Data.Foldable
+import Data.List (sortOn)
+import qualified Data.Map.Strict as Map
+import Data.Ord
+import qualified Data.Vector as Vec
+import Lens.Micro.Platform
+
+import Concordium.Types
+import qualified Concordium.Types.Accounts as Accounts
+import Concordium.Types.BakerIdentity
+import Concordium.Types.Parameters hiding (getChainParameters)
+import Concordium.Utils
+
+import qualified Concordium.Crypto.BlockSignature as Sig
+import Concordium.GlobalState.BakerInfo
+import Concordium.KonsensusV1.TreeState.Implementation
+import Concordium.KonsensusV1.TreeState.Types
+import Concordium.KonsensusV1.Types
+import Concordium.Logger
+import Control.Monad.Reader.Class
+
+-- |A Monad for multicasting timeout messages.
+class MonadMulticast m where
+    -- |Multicast a timeout message.
+    sendTimeoutMessage :: TimeoutMessage -> m ()
+
+    -- |Multicast a quorum message.
+    sendQuorumMessage :: QuorumMessage -> m ()
+
+    -- |Multicast a block.
+    sendBlock :: SignedBlock -> m ()
```

Comment:
Not sure whether it's more niche, but it's more correct as the node only holds onto a subset of peers of the network. 
It's true that, ultimately it will be broadcast onto the network - but that is because of the network topology and how it's implemented - which is not exposed at this level.

But most importantly; it should not give rise to any confusion, so I am renaming it to `MonadBroadcast`